pkg/terraform/parser/parser.go

```diff
@@ -538,3 +538,14 @@ func ParseIndexKey(rs resourceState, is instanceObjectState) (string, error) {
 
 	return strs.Join(".", rs.Module, rs.Type, rs.Name), nil
 }
+
+// IsNumber returns true if the index key is a number.
+// https://github.com/hashicorp/terraform/blob/main/internal/states/statefile/version4.go#L114C1-L122C32.
+// As terraform indexKey treated as int key, assume all number types are int when parsing index key.
+func IsNumber(indexKey any) bool {
+	return reflect.TypeOf(indexKey).Kind() >= reflect.Int && reflect.TypeOf(indexKey).Kind() <= reflect.Float64
```

Comment:
What's the difference comparing to previous implementation? What cases we were missing?

Reply:
It will treat all number type as the case. 
```
package main

import (
	"fmt"
)

func main() {
	var a int = 1
	var b int8 = 2
	var c int16 = 3
	var d int32 = 4
	var e int64 = 5
	var f uint = 6
	var g uint8 = 7
	var h uint16 = 8
	var i uint32 = 9
	var j uint64 = 10
	var k float32 = 11
	var l float64 = 12

	fmt.Printf("a:%v, b:%v, c:%v, d:%v, e:%v, f:%v, g:%v, h:%v, i:%v, j:%v, k:%v, l:%v\n", a, b, c, d, e, f, g, h, i, j, k, l)
}
```

The result will be `a:1, b:2, c:3, d:4, e:5, f:6, g:7, h:8, i:9, j:10, k:11, l:12`, just in case there are other number type. Event though the `encoding/json` will only parse to `float64`